compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -957,6 +963,64 @@ abstract class PrepJSInterop extends plugins.PluginComponent
       case ctor: MethodSymbol if ctor.isPrimaryConstructor => ctor
     }
 
+  private def needsFixOrCleanup(template: Template): Boolean = {
+    val body = template.body
+    if (body == null) {
+      false
+    } else {
+      body.exists { member =>
+        val sym = member.symbol
+        if (sym != null) sym.hasAnnotation(WasPublicBeforeTyperClass)
+        else false
+      }
+    }
+  }
+
+  private def fixPublicBeforeTyper(template: Template): Template = {
+    val body = template.body
+
+    def needsCleanUp(vddef: ValOrDefDef): Boolean =
+      vddef.symbol.hasAnnotation(WasPublicBeforeTyperClass)
+
+    val accessorsNeedFix = body.collect {
+      case vdef: ValDef if needsCleanUp(vdef) => vdef.name.toString.trim
+    }.toSet
+
+    def needsFix(ddef: DefDef): Boolean = {
+      lazy val isAccessorThatNeedsFix = {
+        val accessorName = ddef.name.toString
+        val fieldName = {
+          if (accessorName.endsWith("_$eq"))
+            accessorName.substring(0, accessorName.length - 4)
+          else
+            accessorName
+        }
+        accessorsNeedFix(fieldName)
```

Comment:
Instead of collecting field names, can't we collect the field symbols, and simply use `ddef.symbol.accessed`?